Description: Only create CPDManager if D-Bus descriptions present
Author: Michael Weghorn <m.weghorn@posteo.de>
Upstream Commit: https://gerrit.libreoffice.org/plugins/gitiles/core/+/773e4896521ccf29ab040f977a3faaadd805c8d8%5E%21
Bug: https://launchpad.net/bugs/1798074

diff --git a/vcl/unx/generic/printer/cpdmgr.cxx b/vcl/unx/generic/printer/cpdmgr.cxx
index a830c86c9..2438c98b8 100644
--- a/vcl/unx/generic/printer/cpdmgr.cxx
+++ b/vcl/unx/generic/printer/cpdmgr.cxx
@@ -247,6 +247,12 @@ CPDManager* CPDManager::tryLoadCPD()
     static const char* pEnv = getenv("SAL_DISABLE_CPD");
 
     if (!pEnv || !*pEnv) {
+        // interface description XML files are needed in 'onNameAcquired()'
+        if (!g_file_test(FRONTEND_INTERFACE, G_FILE_TEST_IS_REGULAR) ||
+                !g_file_test(BACKEND_INTERFACE, G_FILE_TEST_IS_REGULAR)) {
+            return nullptr;
+        }
+
         GDir *dir;
         const gchar *filename;
         dir = g_dir_open(BACKEND_DIR, 0, nullptr);
